Source/TestingServices/SchedulingStrategies/Liveness/CycleDetectionStrategy.cs

```diff
@@ -559,16 +583,16 @@ private bool IsSchedulingFair(IEnumerable<ScheduleStep> cycle)
             return result;
         }
 
-        private bool IsNondeterminismFair(IEnumerable<ScheduleStep> cycle)
+        private static bool IsNondeterminismFair(List<ScheduleStep> cycle)
         {
             var fairNondeterministicChoiceSteps = cycle.Where(
                 val => val.Type == (ScheduleStepType.FairNondeterministicChoice) &&
-                val.BooleanChoice != null);
+                val.BooleanChoice != null).ToList();
             foreach (var step in fairNondeterministicChoiceSteps)
             {
-                var choices = fairNondeterministicChoiceSteps.Where(c => c.NondetId.Equals(step.NondetId));
```

Comment:
`fairNondeterministicChoiceSteps` is being enumerated while also creating another enumeration of it. So I added ToList().